### ISSUE TITLE
Broken fix suggestion for no_mangle_with_rust_abi

### ISSUE THREAD

**Issue by balt-dev** [rust-lang/rust-clippy#13656]
### Summary

The fix suggestion for this lint proposes putting a fix where it shouldn't be if there's an occurrence of the string `fn` anywhere before the `fn` keyword.

### Reproducer

```rust
#![warn(clippy::no_mangle_with_rust_abi)]

mod r#fn {
    #[no_mangle]
    pub(in super::r#fn) fn breaking() {}
}
```

This code suggests putting `extern "C"` in-between `r#` and `fn`, instead of `)` and `fn`:
```
warning: `#[no_mangle]` set on a function with the default (`Rust`) ABI
 --> src/lib.rs:5:5
  |
5 |     pub(in super::r#fn) fn breaking() {}
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#no_mangle_with_rust_abi
note: the lint level is defined here
 --> src/lib.rs:1:9
  |
1 | #![warn(clippy::no_mangle_with_rust_abi)]
  |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
help: set an ABI
  |
5 |     pub(in super::r#extern "C" fn) fn breaking() {}
  |                     ++++++++++
help: or explicitly set the default
  |
5 |     pub(in super::r#extern "Rust" fn) fn breaking() {}
  |                     +++++++++++++
```

This is caused by it being implemented with a naive string replacement.

### Version

rustc 1.84.0-nightly (b8c8287a2 2024-11-03)
binary: rustc
commit-hash: b8c8287a229cd79604aa84c25e1235fc78cd5f2e
commit-date: 2024-11-03
host: x86_64-unknown-linux-gnu
release: 1.84.0-nightly
LLVM version: 19.1.3

### Additional Labels

@rustbot label I-suggestion-causes-error

**Comment by balt-dev** [comment]
@nindalf since you were the original implementor, i feel you should know about this

**Comment by balt-dev** [comment]
This _is_ an edge case, but it's still very good to know about.
